### ISSUE TITLE
DBeaver does not export to file if file exists

### ISSUE THREAD

**Issue by ErnieZapata** [dbeaver/dbeaver#21629]
### Description

I just upgraded to the Community version  23.2.3.202310221533 and see that exporting CSV data will not work if the file already exists.

Previously I could export to the same CSV base file name and would see what appears to e have been a timestamp appended to the file name. But now, if the file already exists, nothing is exported, the "Data export task completed" popup on the Windows task bar appears, but the results are not exported.

### DBeaver Version

 23.2.3.202310221533

### Operating System

Windows 10 Professional

### Database and driver

MS SQL Server / SQL Server 

mssql-jdbc-8.2.0.jar

### Steps to reproduce

1. Run a query to produce results
2. Export data to CSV file called 'export' - creates export.csv 
3. Run a different query to produce different results
4. Export data to CSV file called 'export' 

Popup on task bar appears and displays "Data export task completed" but the contents of export.csv does not change. There is an error logged, log file attached.

[log.zip](https://github.com/dbeaver/dbeaver/files/13070985/log.zip)

**Comment by E1izabeth** [comment]
Please show the screenshot of this page of the data transfer wizard
![image](https://github.com/dbeaver/dbeaver/assets/28875055/ed44fe7c-90ce-49fa-bdfe-730fa394d327)

**Comment by ErnieZapata** [comment]
I have never really noticed that section on "File name conflict" as it appears to be collapsed by default. 

Here is the screenshot

![image](https://github.com/dbeaver/dbeaver/assets/20826095/cfdaabf4-186f-4178-8495-da4413a30054)

I changed this option to "Ask" which will work for me, but even if asked and I select "Autofix name", I still see no CSV exported and and I see the same exception and stack trace originally attached

**Comment by E1izabeth** [comment]
I see. Thanks

**Comment by E1izabeth** [comment]
We have already fixed it in #21611 and the fix will be included in DBeaver 23.2.4
For now you could try early access version https://dbeaver.io/files/ea/ - the bug should gone

**Comment by AlterEchoes** [comment]
Issue is still present as of 23.2.4

**Comment by ErnieZapata** [comment]
Confirmed. 

The "Autofix name" on File name conflict still displays "Data export task completed" but does not create a new file or overwrite the existing file with DBeaver 23.2.4. I did not get a chance to test using any preview version.

**Comment by E1izabeth** [comment]
Thanks

**Comment by E1izabeth** [comment]
From log:
```
java.nio.file.NoSuchFileException: C:\Users\user\Album-1699456084038.csv
	at java.base/sun.nio.fs.WindowsException.translateToIOException(WindowsException.java:85)
	at java.base/sun.nio.fs.WindowsException.rethrowAsIOException(WindowsException.java:103)
	at java.base/sun.nio.fs.WindowsException.rethrowAsIOException(WindowsException.java:108)
	at java.base/sun.nio.fs.WindowsFileSystemProvider.newByteChannel(WindowsFileSystemProvider.java:236)
	at java.base/java.nio.file.spi.FileSystemProvider.newOutputStream(FileSystemProvider.java:484)
	at java.base/java.nio.file.Files.newOutputStream(Files.java:228)
	at org.jkiss.dbeaver.tools.transfer.stream.StreamTransferConsumer.openOutputStreams(StreamTransferConsumer.java:506)
	at org.jkiss.dbeaver.tools.transfer.stream.StreamTransferConsumer.initExporter(StreamTransferConsumer.java:385)
	at org.jkiss.dbeaver.tools.transfer.stream.StreamTransferConsumer.fetchStart(StreamTransferConsumer.java:167)
	at org.jkiss.dbeaver.model.impl.jdbc.struct.JDBCTable.readData(JDBCTable.java:195)
	at org.jkiss.dbeaver.tools.transfer.database.DatabaseTransferProducer.lambda$0(DatabaseTransferProducer.java:270)
	at org.jkiss.dbeaver.model.exec.DBExecUtils.tryExecuteRecover(DBExecUtils.java:190)
	at org.jkiss.dbeaver.tools.transfer.database.DatabaseTransferProducer.transferData(DatabaseTransferProducer.java:187)
	at org.jkiss.dbeaver.tools.transfer.database.DatabaseTransferProducer.transferData(DatabaseTransferProducer.java:1)
	at org.jkiss.dbeaver.tools.transfer.DataTransferJob.transferData(DataTransferJob.java:113)
	at org.jkiss.dbeaver.tools.transfer.DataTransferJob.run(DataTransferJob.java:86)
	at org.jkiss.dbeaver.runtime.RunnableContextDelegate.lambda$0(RunnableContextDelegate.java:39)
	at org.eclipse.jface.operation.ModalContext$ModalContextThread.run(ModalContext.java:122)

```

**Comment by ErnieZapata** [comment]
Seems to be resolved with 23.2.5

**Comment by E1izabeth** [comment]
Thank you for update. Confirm that I can't reproduce it anymore too.
